optaplanner-core/src/main/java/org/optaplanner/core/impl/phase/PhaseFactoryProvider.java

```diff
@@ -29,21 +29,26 @@
 import org.optaplanner.core.impl.partitionedsearch.DefaultPartitionedSearchPhaseFactory;
 import org.optaplanner.core.impl.phase.custom.DefaultCustomPhaseFactory;
 
-public class PhaseFactoryProvider<Solution_> {
+public class PhaseFactoryProvider {
 
-    public AbstractPhaseFactory createPhaseFactory(PhaseConfig<?> phaseConfig) {
+    private PhaseFactoryProvider() {
+        throw new UnsupportedOperationException(
+                String.format("The class (%s) should not be instantiated.", PhaseFactoryProvider.class));
+    }
+
+    public static <Solution_> PhaseFactory<Solution_> createPhaseFactory(PhaseConfig<?> phaseConfig) {
```

Comment:
This should be a factory method on the `PhaseFactory` interface. Same pattern as with `SolutionFactory`.